### PR TITLE
cleanup python modular graphical examples

### PR DESCRIPTION
**Pull request by sanuj**
Most of the examples were working except the following:
`so_multiclass_director_BMRM.py`:
```
ImportError: cannot import name DirectorStructuredModel
```
`so_multiclass_BMRM.py`:
```
Traceback (most recent call last):
  File "so_multiclass_BMRM.py", line 78, in <module>
    plt.plot(xrange(res.get_n_iters()), Fps, hold=True)
  File "/usr/local/lib/python2.7/dist-packages/matplotlib/pyplot.py", line 3154, in plot
    ret = ax.plot(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/matplotlib/__init__.py", line 1812, in inner
    return func(ax, *args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/matplotlib/axes/_axes.py", line 1424, in plot
    for line in self._get_lines(*args, **kwargs):
  File "/usr/local/lib/python2.7/dist-packages/matplotlib/axes/_base.py", line 386, in _grab_next_args
    for seg in self._plot_args(remaining, kwargs):
  File "/usr/local/lib/python2.7/dist-packages/matplotlib/axes/_base.py", line 364, in _plot_args
    x, y = self._xy_from_xy(x, y)
  File "/usr/local/lib/python2.7/dist-packages/matplotlib/axes/_base.py", line 223, in _xy_from_xy
    raise ValueError("x and y must have same first dimension")
ValueError: x and y must have same first dimension
```
`regression_gaussian_process_demo.py` and `regression_gaussian_process_modelselection.py`:
```
Traceback (most recent call last):
  File "regression_gaussian_process_demo.py", line 14, in <module>
    from modshogun import GradientModelSelection
ImportError: cannot import name GradientModelSelection
```
`eigenfaces.py`:
```
Reading images ...
Traceback (most recent call last):
  File "eigenfaces.py", line 184, in <module>
    images = readImages(list_filenames);
  File "eigenfaces.py", line 132, in readImages
    images = np.empty( (IMAGE_HEIGHT*IMAGE_WIDHT, (len(list_filenames))-1))
ValueError: negative dimensions are not allowed
```
`interactive_gp_demo.py`:
```
Traceback (most recent call last):
  File "interactive_gp_demo.py", line 218, in plot_gp
    inf.get_negative_marginal_likelihood()
AttributeError: 'ExactInferenceMethod' object has no attribute 'get_negative_marginal_likelihood'
```
`metric_lmnn_objective.py`:
```
IOError: [Errno 2] No such file or directory: '../../data/fm_train_multiclass_digits.dat.gz'
```

### CODE REVIEW

**Comment by sanuj** [comment]
I'm cloning shogun-data right now. Will run `metric_lmnn_objective.py` and `eigenfaces.py` again after that.

File: examples/undocumented/python_modular/graphical/interactive_clustering_demo.py [hunk 1]
```diff
@@ -11,10 +11,9 @@
 from PyQt4.QtGui import *
 
 import matplotlib
-from matplotlib import mpl
```
Comment (sanuj) [inline examples/undocumented/python_modular/graphical/interactive_clustering_demo.py]: gives import error so removing this.

File: examples/undocumented/python_modular/graphical/interactive_clustering_demo.py [hunk 2]
```diff
@@ -11,10 +11,9 @@
...
 from matplotlib.colorbar import make_axes, Colorbar
 from matplotlib.backends.backend_qt4agg import FigureCanvasQTAgg as FigureCanvas
-from matplotlib.backends.backend_qt4agg import NavigationToolbar2QTAgg as NavigationToolbar
```
Comment (sanuj) [inline examples/undocumented/python_modular/graphical/interactive_clustering_demo.py]: `NavigationToolbar2QTAgg` replaced by `NavigationToolbar2QT` as the former gives import error.

File: examples/undocumented/python_modular/graphical/interactive_svm_demo.py [hunk 3]
```diff
@@ -132,7 +129,7 @@ def train_svm(self):
         svm.set_epsilon(1e-2)
 
         x, y, z = util.compute_output_plot_isolines(svm, gk, train)
-        plt=self.axes.pcolor(x, y, z, shading='interp')
```
Comment (sanuj) [inline examples/undocumented/python_modular/graphical/interactive_svm_demo.py]: `shading` is deprecated in `matplotlib` so removing this parameter. No visual changes in the plots.
```
/usr/local/lib/python2.7/dist-packages/matplotlib/cbook.py:137: MatplotlibDeprecationWarning: The shading option was deprecated in version 1.2. Use edgecolors instead.
  warnings.warn(message, mplDeprecation, stacklevel=1)
```

**Comment by sanuj** [comment]
I found these bugs while I was playing with `KernelRidgeRegression` python example and then decided to clean all the graphical examples. I must add, the interactive examples (`interactive_svm_demo.py` etc) are fun to play with :)

**Comment by sanuj** [comment]
`eigenfaces.py` and `metric_lmnn_objective.py` work with shogun-data.
@karlnapf Please review.

**Comment by karlnapf** [comment]
Nice thanks for that.

**Comment by karlnapf** [comment]
BTW we should include these graphical examples in the build, will open an issue

**Pull request merged**